packages/google_sign_in/google_sign_in_web/lib/src/button_configuration.dart

```diff
@@ -0,0 +1,202 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:google_identity_services_web/id.dart' as id;
+import 'package:js/js_util.dart' as js_util;
+
+/// Converts user-facing `GisButtonConfiguration` into the JS-Interop `id.GsiButtonConfiguration`.
+id.GsiButtonConfiguration? convertButtonConfiguration(
+  GSIButtonConfiguration? config,
+) {
+  if (config == null) {
+    return null;
+  }
+  return js_util.jsify(<String, Object?>{
+    if (config.type != null) 'type': _idType[config.type],
+    if (config.theme != null) 'theme': _idTheme[config.theme],
+    if (config.size != null) 'size': _idSize[config.size],
+    if (config.text != null) 'text': _idText[config.text],
+    if (config.shape != null) 'shape': _idShape[config.shape],
+    if (config.logoAlignment != null)
+      'logo_alignment': _idLogoAlignment[config.logoAlignment],
+    if (config.minimumWidth != null) 'width': config.minimumWidth,
+    if (config.locale != null) 'locale': config.locale,
+  }) as id.GsiButtonConfiguration;
+}
+
+/// A class to configure the Google Sign-In Button for web.
+///
+/// https://developers.google.com/identity/gsi/web/reference/js-reference#GsiButtonConfiguration
```

Comment:
nit: Flutter style typically wants something like this:

```
/// See also:
///   * link
```

This comment applies throughout the file.

Reply:
Done.